apache/metron
# METRON-1774 Allow user to configure JAAS client in Ambari

Right now, the user can only configure Metron's client JAAS in limited ways.  They can configure the principal name and key tab path, but few other things.  For example, I cannot easily turn on debugging today.  If I make a change to `$METRON_HOME/client_jaas.conf` to set `debug=true`, Ambari will overwrite my change when any of the topologies are restarted.

With this PR the user can now edit a template in Ambari that is used to generate `$METRON_HOME/client_jaas.conf`.  This allows user to turn on debug logs, change the service name, or add additional options required by their environment.

![screen shot 2018-09-11 at 2 12 35 pm](https://user-images.githubusercontent.com/2475409/45381641-d9b45080-b5d3-11e8-9afe-afd79c61ad4c.png)

## Testing

1. Stand-up a development environment.

1. Ensure alerts are visible in the Alerts UI.

1. Kerberize that development environment.

1. Ensure alerts are visible in the Alerts UI.

1. Ensure that Metron's `client_jaas.conf` was created correctly.

    ```
    [root@node1 0.6.0]# source /etc/default/metron
    [root@node1 0.6.0]# cat $METRON_HOME/client_jaas.conf

    StormClient {
       com.sun.security.auth.module.Krb5LoginModule required
       useKeyTab=false
       storeKey=false
       useTicketCache=true
       serviceName="nimbus"
       principal="metron@EXAMPLE.COM";
    };
    Client {
       com.sun.security.auth.module.Krb5LoginModule required
       useKeyTab=true
       keyTab="/etc/security/keytabs/metron.headless.keytab"
       storeKey=true
       useTicketCache=false
       serviceName="zookeeper"
       principal="metron@EXAMPLE.COM";
    };
    KafkaClient {
       com.sun.security.auth.module.Krb5LoginModule required
       useKeyTab=true
       keyTab="/etc/security/keytabs/metron.headless.keytab"
       storeKey=true
       useTicketCache=false
       serviceName="kafka"
       principal="metron@EXAMPLE.COM";
    ```

1. In Ambari, go to Metron > Configs > Advanced > Advanced metron-client-jaas-conf and alter the template in some way.  For example, you could add `debug=true` to each stanza.

1. Restart the Enrichment topology.

1. Validate that Metron's `client_jaas.conf` was updated based on the changes you made in Ambari.

    ```
    [root@node1 0.6.0]# cat client_jaas.conf
    StormClient {
       com.sun.security.auth.module.Krb5LoginModule required
       debug=true
       useKeyTab=false
       storeKey=false
       useTicketCache=true
       serviceName="nimbus"
       principal="metron@EXAMPLE.COM";
    };
    Client {
       com.sun.security.auth.module.Krb5LoginModule required
       debug=true
       useKeyTab=true
       keyTab="/etc/security/keytabs/metron.headless.keytab"
       storeKey=true
       useTicketCache=false
       serviceName="zookeeper"
       principal="metron@EXAMPLE.COM";
    };
    KafkaClient {
       com.sun.security.auth.module.Krb5LoginModule required
       debug=true
       useKeyTab=true
       keyTab="/etc/security/keytabs/metron.headless.keytab"
       storeKey=true
       useTicketCache=false
       serviceName="kafka"
       principal="metron@EXAMPLE.COM";
    ```

## Pull Request Checklist
